### PR TITLE
Feature/#1 convert to typescript the authjs file

### PR DESCRIPTION
**Pull request by DavielSA**

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[0820/175240.354:ERROR:crash_report_database_win.cc(580)] CreateDirectory C:\Program Files (x86)\Cuppat\Reports\Dump: El sistema no puede encontrar la ruta especificada. (0x3)
```
Comment (dailymp) [inline debug.log]: This file should be removed and added to gitignore

File: src/Auth/Auth.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  private _scopes = null;
+  private _expiresAt:number = 0;
+  private userProfile:any;
+  private requestedScopes:string;
+  private auth0:auth0.WebAuth;
+  private history:any;
+
+  constructor(history:any) {
+    this.history = history;
+    //this.history = history;
```
Comment (dailymp) [inline src/Auth/Auth.ts]: Remove this code commented

File: src/Auth/Auth.ts [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+    options.redirectUri = process.env.REACT_APP_AUTH0_CALLBACK_URL;
+    options.audience = process.env.REACT_APP_AUTH0_AUDIENCE ? process.env.REACT_APP_AUTH0_AUDIENCE : "";
+    options.responseType ="token id_token";
+    options.scope =this.requestedScopes;
+
+    this.auth0 = new auth0.WebAuth(options);
+  }
+
+  public login = () => {
+    localStorage.setItem(
```
Comment (dailymp) [inline src/Auth/Auth.ts]: Which type return this funtion? Can you please infer and type it?

File: src/Auth/Auth.ts [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      this.REDIRECT_ON_LOGIN,
+      JSON.stringify(this.history.location)
+    );
+    this.auth0.authorize();
+  }
+
+  private handleAuthentication = () => {
```
Comment (dailymp) [inline src/Auth/Auth.ts]: If you can you should type all the functions types!! Please! Thanks!

File: src/Auth/Auth.ts [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      } else if (err) {
+        this.history.push("/");
+        alert(`Error: ${err.error}. Check the console for further details.`);
+        console.log(err);
+      }
+      localStorage.removeItem(this.REDIRECT_ON_LOGIN);
+    });
+  };
+
+  private setSession = (authResult:any) => {
```
Comment (dailymp) [inline src/Auth/Auth.ts]: The same than previous function

File: src/Auth/Auth.ts [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+    console.log(authResult);
+    // set the time that the access token will expire
```
Comment (dailymp) [inline src/Auth/Auth.ts]: You must removed all code commented and debuggers, console.log etc when doing Pull Request

File: debug.log (hunk 1 above)
Comment (DavielSA) [reply debug.log]: OK. Done.

File: src/Auth/Auth.ts (hunk 2 above)
Comment (DavielSA) [reply src/Auth/Auth.ts]: OK. Done.

File: src/Auth/Auth.ts (hunk 3 above)
Comment (DavielSA) [reply src/Auth/Auth.ts]: OK. Done.

File: src/Auth/Auth.ts (hunk 4 above)
Comment (DavielSA) [reply src/Auth/Auth.ts]: OK. Done.

File: src/Auth/Auth.ts (hunk 5 above)
Comment (DavielSA) [reply src/Auth/Auth.ts]: OK. Done.

File: src/Auth/Auth.ts (hunk 6 above)
Comment (DavielSA) [reply src/Auth/Auth.ts]: OK. Done.

**Pull request merged**